### PR TITLE
fix(rust): fix conversion from decimal to float

### LINKED ISSUE #9873: Decimal value parsed into incorrect float value
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python

from decimal import Decimal
import polars as pl

data = [
    ('23851007560160171', 'L1', '361359502489563', Decimal('45.0000'), Decimal('0')), 
    ('23851007560260171', 'L1', '361359502489563', Decimal('45.0000'), Decimal('0'))
]
df = pl.from_records(data, orient='row')
print(df)

# Output:
# shape: (2, 5)
# ┌───────────────────┬──────────┬─────────────────┬──────────┬──────────┐
# │ column_0          ┆ column_1 ┆ column_2        ┆ column_3 ┆ column_4 │
# │ ---               ┆ ---      ┆ ---             ┆ ---      ┆ ---      │
# │ str               ┆ str      ┆ str             ┆ f64      ┆ f64      │
# ╞═══════════════════╪══════════╪═════════════════╪══════════╪══════════╡
# │ 23851007560160171 ┆ L1       ┆ 361359502489563 ┆ 450000.0 ┆ 0.0      │
# │ 23851007560260171 ┆ L1       ┆ 361359502489563 ┆ 450000.0 ┆ 0.0      │
# └───────────────────┴──────────┴─────────────────┴──────────┴──────────┘

```

### Issue description

`Decimal` value is not getting parsed into incorrect float value e.g. `Decimal('45.0000') -> 450000.0` , `Decimal('45.00') -> 4500.0` , etc.

### Expected behavior

Should get parsed into correct float value e.g. `Decimal('45.0000') -> 45.0` , `Decimal('45.00') -> 45.0` , etc.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.7
Index type:          UInt32
Platform:            Linux-5.19.0-46-generic-x86_64-with-glibc2.35
Python:              3.11.4 (main, Jun  7 2023, 12:45:48) [GCC 11.3.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
matplotlib:          <not installed>
numpy:               1.24.3
pandas:              2.0.2
pyarrow:             11.0.0
pydantic:            1.10.7
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>

```

</details>

### PR DESCRIPTION
**Pull request by c-peters**
fix #9873

### CODE REVIEW

File: py-polars/tests/unit/datatypes/test_decimal.py [hunk 1]
```diff
@@ -84,6 +84,15 @@ def test_init_decimal_dtype() -> None:
     assert df["a"].is_numeric()
 
 
+def test_decimal_cast_by_schema(float_tolerance: float = 0.01) -> None:
+    df = pl.DataFrame(
+        {"a": [D("1"), D("2.55"), D("45.000"), D("10.0")]}, schema={"a": pl.Float64}
+    )
+    assert (
+        abs(df["a"].sum() - 58.55) < float_tolerance
+    ), "Something went wrong converting decimals to floats"
```
Comment (stinodego) [inline py-polars/tests/unit/datatypes/test_decimal.py:93]: We don't really need assert messages, as the test name already documents this for us.

I would write the assertion like this:

```suggestion
    expected = pl.DataFrame({"a": [1.0, 2.55, 45.0, 10.0]})
    assert_frame_equal(df, expected)
```

Our assertion utils have float tolerance built-in.

**Review by stinodego — CHANGES_REQUESTED** [review]
The fix seems to work! I just have a comment about the test, otherwise the implementation seems fine to me.

File: py-polars/tests/unit/datatypes/test_decimal.py (hunk 1 above)
Comment (c-peters) [reply py-polars/tests/unit/datatypes/test_decimal.py]: Thanks @stinodego !

Comment (stinodego) [reply py-polars/tests/unit/datatypes/test_decimal.py]: I'm not sure what caused the original bug, but your test still doesn't properly test for it. Your horizontal example also passes on `main`.

I'd say always add the minimal reproducible example from the original issue as a test, and we often include the issue number in the test name as well.

Comment (stinodego) [reply py-polars/tests/unit/datatypes/test_decimal.py]: I went ahead and added the test myself!

**Review by stinodego — APPROVED** [review]

**Comment by stinodego** [comment]
Looks good to me! Maybe someone with more knowledge of how decimals work / Rust knowhow can take a second look.

**Review by orlp — APPROVED** [review]

**Pull request merged**